### PR TITLE
fix: add type check to pattern.find

### LINKED ISSUE #101: "Error executing vim.schedule lua callback: Vim:E976: Using a Blob as a String" with omni completion
### Not a question

- [X] This is not a question. If you have a question, use [discussions](https://github.com/uga-rosa/ccc.nvim/discussions).

### I investigated.

- [X] I have read README.md, help documentation, and searched past issues but no solution.

### Description

When using omni completion via the builtin `i_CTRL-X_CTRL-O` command, the following error message is generated by this plugin:
```
Error executing vim.schedule lua callback: Vim:E976: Using a Blob as a String
stack traceback:
        [C]: in function 'matchlist'
        ...local/share/nvim/lazy/ccc.nvim/lua/ccc/utils/pattern.lua:37: in function 'find'
        ...o/.local/share/nvim/lazy/ccc.nvim/lua/ccc/picker/hex.lua:36: in function 'parse_color'
        .../.local/share/nvim/lazy/ccc.nvim/lua/ccc/highlighter.lua:311: in function 'update_picker'
        .../.local/share/nvim/lazy/ccc.nvim/lua/ccc/highlighter.lua:163: in function <.../.local/share/nvim/lazy/ccc.nvim/lua/ccc/highlighter.lua:161>
```

### Environments

`nvim` version: v0.10.0-dev-2384+g848fc8ede
OS: Both reproducible in macOS 14.3.1 (23D60) and Fedora Asahi Remix

### Minimal reproducible full config

_I'm sorry in advance for not using the provided template. I'm VERY unfamiliar with Vimscript, and I had lots of trouble building a proper minimal configuration for this issue. Yet I thrive on submitting high-quality and helpful issues, hence the configuration below is written in Lua (please pardon my skill issues) and has been carefully tested to truly represent a minimal setup. If this represents an issue for you, please lmk and I'll accommodate._

```lua
local plugins = {
  ccc = 'https://github.com/uga-rosa/ccc.nvim.git',
  lspconfig = 'https://github.com/neovim/nvim-lspconfig.git',
  lsp_compl = 'https://github.com/mfussenegger/nvim-lsp-compl.git',
}

-- Make sure that the executable for lua_lsp in available in your path!
-- E.g.: When installing `lua_ls` with Mason: vim.env.PATH = vim.env.PATH .. ':' .. vim.fn.stdpath 'data' .. '/mason/bin'

for name, url in pairs(plugins) do
  local install_path = 'repro/' .. name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
  vim.o.runtimepath = install_path .. ',' .. vim.o.runtimepath
end

require'lspconfig'.lua_ls.setup { on_attach = require 'lsp_compl'.attach }
```

### Steps to reproduce

1. Inside a folder called `repro` create the provided `nvim` configuration file called `minimal.lua`.
2. Run `nvim --clean -u minimal.lua minimal.lua -c "normal10G"`.
3. Type the following commands: `o<Ctrl-x><Ctrl-o><Tab><Ctrl-x><Ctrl-o>`.
4. Observe the generated error message.

### Expected behavior

For this plugin to not affect omni completion.

### Actual behavior

A disruptive error message affects the omni completion experience.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MariaSolOs**
Fixes #101

### CODE REVIEW

File: lua/ccc/utils/pattern.lua [hunk 1]
```diff
@@ -34,6 +34,11 @@ end
 ---@return integer? end
 ---@return string? ... submatches
 function pattern.find(str, pat, init)
+  -- Avoid vim:E976 error when the given string has a blob.
+  if vim.fn.type(str) ~= vim.v.t_string then
+    return
+  end
```
Comment (uga-rosa) [inline lua/ccc/utils/pattern.lua:40]: A more rigorous form, avoiding calls to vim functions, is preferred.
```lua
-- matchlist() considers a string containing `\n` as a blob and cannot process them.
if str:find("\n") then
  return
end
```

**Review by uga-rosa — CHANGES_REQUESTED** [review]

File: lua/ccc/utils/pattern.lua (hunk 1 above)
Comment (MariaSolOs) [reply lua/ccc/utils/pattern.lua]: @uga-rosa unfortunately this change doesn't solve the issue (see my comment [here](https://github.com/uga-rosa/ccc.nvim/issues/101#issuecomment-1977910652)).

If using the code I originally proposed, then the issue _is_ truly fixed.

Comment (uga-rosa) [reply lua/ccc/utils/pattern.lua]: Oh, sorry. This was a misunderstanding. It's a null character, not an `\n`.

Comment (uga-rosa) [reply lua/ccc/utils/pattern.lua]: fixed in 86299a4562c86c29295361721f7229a0b99a17b8.

Comment (MariaSolOs) [reply lua/ccc/utils/pattern.lua]: Thank you!

**Pull request merged**